src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,50 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  static MIN_HEALTH = 1;
+
+  health = 100;
+
+  static removeFromAlive() {
+    this.alive = this.alive.filter((item) => item.health > this.MIN_HEALTH);
+  }
+
+  constructor(name) {
+    this.name = name;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+
+  bite(animal) {
+    if (animal instanceof Carnivore || animal.hidden) {
+      return;
+    }
+
+    animal.health -= 50;
+
+    if (animal.health < Animal.MIN_HEALTH) {
```

Comment:
This logic to remove dead animals is inefficient as it might call `Animal.removeFromAlive` multiple times if multiple animals are bitten in a sequence. It would be better to call this method once after all the biting actions are done, perhaps outside of the `bite` method.